Problems with using ulid with `mypy --strict`

Hi,
my project is using `mypy --strict`.
While importing ulid I'm getting a problem:

```python
import ulid

MY_ULID = ulid.new()
```

```text
error: Module has no attribute "new"
```

I found a workaround:
```python
import ulid

MY_ULID = ulid.api.new()
```

But I'm sure the first way is a bit more preferable.

## Investigation

I made some investigation on the problem.
The following modified content of `__init__.py` should fix the problem:

```python
from .api import from_bytes, from_int, from_randomness, from_str, from_timestamp, from_uuid, new, parse
from .ulid import Randomness, Timestamp, ULID


__all__ = [
    # from .api
    'new', 'parse', 'from_bytes', 'from_int', 'from_str', 'from_uuid', 'from_timestamp', 'from_randomness',
    # from .ulid
    'Timestamp', 'Randomness', 'ULID',
]

__version__ = '0.0.14' 
```

So I explicitly imported items and explicitly listed them in `__all__ `. This is some code duplication, but it looks not fatal for me.

## Questions

Q1. Should I create PR with the these changes for in `__init__.py`?

Q2. Should I crate PR to fix all `mypy --strict` errors for the whole ulid project? The fixes are going to be trivial from my experience. Here is the full list of mypy errors:

```text
ulid\ulid.py:23: error: Function is missing a type annotation
ulid\ulid.py:26: error: Function is missing a type annotation
ulid\ulid.py:39: error: Function is missing a type annotation
ulid\ulid.py:52: error: Function is missing a type annotation
ulid\ulid.py:67: error: Function is missing a type annotation
ulid\ulid.py:82: error: Function is missing a type annotation
ulid\ulid.py:97: error: Function is missing a type annotation
ulid\ulid.py:112: error: Function is missing a return type annotation
ulid\ulid.py:115: error: Function is missing a return type annotation
ulid\ulid.py:118: error: Function is missing a return type annotation
ulid\ulid.py:121: error: Function is missing a return type annotation
ulid\ulid.py:124: error: Function is missing a return type annotation
ulid\ulid.py:127: error: Function is missing a return type annotation
ulid\ulid.py:275: error: Returning Any from function declared to return "Timestamp"
ulid\ulid.py:275: error: Call to untyped function "Timestamp" in typed context
ulid\ulid.py:284: error: Returning Any from function declared to return "Randomness"
ulid\ulid.py:284: error: Call to untyped function "Randomness" in typed context
ulid\api.py:47: error: Returning Any from function declared to return "ULID"
ulid\api.py:47: error: Call to untyped function "ULID" in typed context
ulid\api.py:104: error: Returning Any from function declared to return "ULID"
ulid\api.py:104: error: Call to untyped function "ULID" in typed context
ulid\api.py:124: error: Returning Any from function declared to return "ULID"
ulid\api.py:124: error: Call to untyped function "ULID" in typed context
ulid\api.py:137: error: Returning Any from function declared to return "ULID"
ulid\api.py:137: error: Call to untyped function "ULID" in typed context
ulid\api.py:149: error: Returning Any from function declared to return "ULID"
ulid\api.py:149: error: Call to untyped function "ULID" in typed context
ulid\api.py:198: error: Returning Any from function declared to return "ULID"
ulid\api.py:198: error: Call to untyped function "ULID" in typed context
ulid\api.py:244: error: Returning Any from function declared to return "ULID"
ulid\api.py:244: error: Call to untyped function "ULID" in typed context
```

--- a/Makefile
+++ b/Makefile
@@ -60,7 +60,7 @@ isort:  ## Run isort on the package.
 
 .PHONY: mypy
 mypy:  ## Run mypy static analysis checks on the package.
-	@mypy ulid
+	@mypy ulid --strict
 
 .PHONY: pylint
 pylint:  ## Run pylint on the package.

--- a/ulid/hints.py
+++ b/ulid/hints.py
@@ -8,6 +8,9 @@ import datetime
 import typing
 import uuid
 
+#: Type hint that is an alias for the built-in :class:`~bool` type.
+Bool = bool
+
 #: Type hint that defines multiple types that implement the buffer protocol
 #: that can encoded into a Base32 string.
 Buffer = typing.Union[bytes, bytearray, memoryview]  # pylint: disable=invalid-name
@@ -25,12 +28,17 @@ Int = int  # pylint: disable=invalid-name
 Float = float  # pylint: disable=invalid-name
 
 
+#: Type hint that defines multiple primitive types that can represent parts or full ULID.
+Primitive = typing.Union[int, float, str, bytes, bytearray, memoryview]  # pylint: disable=invalid-name
+
+
 #: Type hint that is an alias for the built-in :class:`~str` type.
 Str = str  # pylint: disable=invalid-name
 
 
 #: Type hint that is an alias for the built-in :class:`~datetime.datetime` type.
 Datetime = datetime.datetime  # pylint: disable=invalid-name
+
 
 #: Type hint that is an alias for the built-in :class:`~datetime.datetime` type.
 UUID = uuid.UUID  # pylint: disable=invalid-name

--- a/ulid/ulid.py
+++ b/ulid/ulid.py
@@ -5,11 +5,16 @@
     Object representation of a ULID.
 """
 import datetime
+import typing
 import uuid
 
 from . import base32, hints
 
 __all__ = ['Timestamp', 'Randomness', 'ULID']
+
+
+#: Type hint that defines multiple primitive types and itself for comparing MemoryView instances.
+MemoryViewPrimitive = typing.Union['MemoryView', hints.Primitive]  # pylint: disable=invalid-name
 
 
 class MemoryView:
@@ -20,10 +25,10 @@ class MemoryView:
 
     __slots__ = ['memory']
 
-    def __init__(self, buffer):
+    def __init__(self, buffer: hints.Buffer) -> None:
         self.memory = memoryview(buffer)
 
-    def __eq__(self, other):
+    def __eq__(self, other: MemoryViewPrimitive) -> hints.Bool:  # type: ignore[override]
         if isinstance(other, MemoryView):
             return self.memory == other.memory
         if isinstance(other, (bytes, bytearray, memoryview)):
@@ -36,7 +41,7 @@ class MemoryView:
             return self.str == other
         return NotImplemented
 
-    def __ne__(self, other):
+    def __ne__(self, other: MemoryViewPrimitive) -> hints.Bool:  # type: ignore[override]
         if isinstance(other, MemoryView):
             return self.memory != other.memory
         if isinstance(other, (bytes, bytearray, memoryview)):
@@ -49,7 +54,7 @@ class MemoryView:
             return self.str != other
         return NotImplemented
 
-    def __lt__(self, other):
+    def __lt__(self, other: MemoryViewPrimitive) -> hints.Bool:
         if isinstance(other, MemoryView):
             return self.int < other.int
         if isinstance(other, (bytes, bytearray)):
@@ -64,7 +69,7 @@ class MemoryView:
             return self.str < other
         return NotImplemented
 
-    def __gt__(self, other):
+    def __gt__(self, other: MemoryViewPrimitive) -> hints.Bool:
         if isinstance(other, MemoryView):
             return self.int > other.int
         if isinstance(other, (bytes, bytearray)):
@@ -79,7 +84,7 @@ class MemoryView:
             return self.str > other
         return NotImplemented
 
-    def __le__(self, other):
+    def __le__(self, other: MemoryViewPrimitive) -> hints.Bool:
         if isinstance(other, MemoryView):
             return self.int <= other.int
         if isinstance(other, (bytes, bytearray)):
@@ -94,7 +99,7 @@ class MemoryView:
             return self.str <= other
         return NotImplemented
 
-    def __ge__(self, other):
+    def __ge__(self, other: MemoryViewPrimitive) -> hints.Bool:
         if isinstance(other, MemoryView):
             return self.int >= other.int
         if isinstance(other, (bytes, bytearray)):
@@ -109,22 +114,22 @@ class MemoryView:
             return self.str >= other
         return NotImplemented
 
-    def __hash__(self):
+    def __hash__(self) -> hints.Int:
         return hash(self.memory)
 
-    def __bytes__(self):
+    def __bytes__(self) -> hints.Bytes:
         return self.bytes
 
-    def __float__(self):
+    def __float__(self) -> hints.Float:
         return self.float
 
-    def __int__(self):
+    def __int__(self) -> hints.Int:
         return self.int
 
-    def __repr__(self):
+    def __repr__(self) -> hints.Str:
         return '<{}({!r})>'.format(self.__class__.__name__, str(self))
 
-    def __str__(self):
+    def __str__(self) -> hints.Str:
         return self.str
 
     @property

